### PR TITLE
fix: add terraform failure if reset api key fails to create api key

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Issue: https://github.ibm.com/GoldenEye/issues/issues/11046

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: scripts/reset_iks_api_key.sh [hunk 1]
```diff
@@ -54,14 +54,25 @@ if [ "${reset}" == true ]; then
     if [ "$PRIVATE_ENV" = true ]; then
         if [ "$CLUSTER_ENDPOINT" == "private" ] || [ "$CLUSTER_ENDPOINT" == "default" ]; then
             RESET_URL="https://private.$REGION.containers.cloud.ibm.com/v1/keys"
-            curl -H "accept: application/json" -H "Authorization: $IAM_TOKEN" -H "X-Auth-Resource-Group: $RESOURCE_GROUP_ID" -X POST "$RESET_URL"
+            result=$(curl -i -H "accept: application/json" -H "Authorization: $IAM_TOKEN" -H "X-Auth-Resource-Group: $RESOURCE_GROUP_ID" -X POST "$RESET_URL" 2>/dev/null)
+            status_code=$(echo "$result" | head -n 1 | cut -d$' ' -f2)
```
Comment (ocofaigh) [inline scripts/reset_iks_api_key.sh:58]: This doesn't seem very clean. `curl` has a built in way to fail if the response code is a failure response using the `--fail` flag. Take a look at https://curl.se/docs/manpage.html

Comment (Aashiq-J) [reply scripts/reset_iks_api_key.sh:58]: This is the current output after the change if the reset fails:
Screenshot 2024-10-09 at 3 12 45 PM

This is the error when we use `--fail` and `--show-error`
Screenshot 2024-10-09 at 3 32 11 PM

Comment (ocofaigh) [reply scripts/reset_iks_api_key.sh:58]: ok this output looks good

**Review by ocofaigh — APPROVED** [review]

**Comment by Aashiq-J** [comment]
Updating the cluster autoscaler in this PR itself, so have to skip upgrade test.
Screenshot 2024-10-15 at 1 56 34 PM

**Pull request merged**